RUFAS/routines/animal/ration/ration_driver.py

```diff
@@ -309,64 +326,580 @@ class RationReporter:
+
+        Parameters
+        ----------
+        kg_fed : float
+            Kilograms of feed item in ration
+        feed_item_info : Dict
+            Dictionary of nutrient and energy information of feed item
+        ration_report : Dict
+            Dictionary of nutrient amount and concentrations
+        body_weight : float
+            Animal body weight in kg
+
+        Returns
+        -------
+        float
+            Fat supply of feed item, g TODO: check units
+
+        """
+        fat_item = feed_item_info["EE"] * kg_fed
+        return fat_item
+
+    @classmethod
+    def get_fat_percentage(cls, kg_fed, feed_item_info, ration_report, body_weight):
+        """
+        Returns fat content percentage of feed item
+
+        Parameters
+        ----------
+        kg_fed : float
+            Kilograms of feed item in ration
+        feed_item_info : Dict
+            Dictionary of nutrient and energy information of feed item
+        ration_report : Dict
+            Dictionary of nutrient amount and concentrations
+        body_weight : float
+            Animal body weight in kg
+
+        Returns
+        -------
+        float
+            Dry matter fat percentage of feed item
+
+        """
+        fat_item_percentage = (feed_item_info["EE"] * kg_fed) / ration_report["nutrient_amount"]["dm"]
+        return fat_item_percentage
+
+    @classmethod
+    def get_forage_NDF(cls, kg_fed, feed_item_info, ration_report, body_weight):
+        """
+        Returns forage neutral detergent fiber content of feed item, kg
+
+        Parameters
+        ----------
+        kg_fed : float
+            Kilograms of feed item in ration
+        feed_item_info : Dict
+            Dictionary of nutrient and energy information of feed item
+        ration_report : Dict
+            Dictionary of nutrient amount and concentrations
+        body_weight : float
+            Animal body weight in kg
+
+        Returns
+        -------
+        float
+            Forage neutral detergent fiber content of feed item, TODO: check units
+
+        """
+        if feed_item_info["type"] == "Forage":
+            forage_NDF_item = feed_item_info["NDF"] * kg_fed
+        else:
+            forage_NDF_item = 0.0
+        return forage_NDF_item
+
+    @classmethod
+    def get_metabolizable_protein(cls, ration, available_feeds, ration_report, body_weight):
+        """
+        Returns metabolizable protein supply  of feed item, TODO: check units
+
+        Parameters
+        ----------
+        kg_fed : float
+            Kilograms of feed item in ration
+        feed_item_info : Dict
+            Dictionary of nutrient and energy information of feed item
+        ration_report : Dict
+            Dictionary of nutrient amount and concentrations
+        body_weight : float
+            Animal body weight in kg
+
+        Returns
+        -------
+        float
+            Metabolizable protein supply of feed item, TODO: check units
```

Comment:
```suggestion
        Returns metabolizable protein supply  of feed item, TODO: check units.

        Parameters
        ----------
        kg_fed : float
            Kilograms of feed item in ration.
        feed_item_info : Dict
            Dictionary of nutrient and energy information of feed item.
        ration_report : Dict
            Dictionary of nutrient amount and concentrations.
        body_weight : float
            Animal body weight in kg.

        Returns
        -------
        float
            Metabolizable protein supply of feed item, TODO: check units.
```
Same comments as above for TODO and top level docstring.